File: programs-basic-java/src/main/java/com/local/numbertypes/NelsonNumber.java
package com.local.numbertypes;

import lombok.extern.slf4j.Slf4j;

@Slf4j
public class NelsonNumber implements INumberTypeOperation {
	// The number 111 is sometimes called a Nelson. Multiples of 111 are called a
	// double Nelson (222), triple Nelson (333)
	// 111,222,333,444,555,666,777

	@Override
	public boolean execute(long input, long order) {
		var status = (input % 111 == 0);
		log(status, input);
		return status;
	}

	private void log(boolean status, long input) {
		if (status) {
			log.debug("An Nelson Number {}", input);
		} else {
			log.debug("Not an Nelson Number {}", input);
		}
	}

}

File: programs-basic-kotlin/src/main/kotlin/com/local/numbertypes/NelsonNumber.kt
package com.local.numbertypes

import mu.KotlinLogging
private val logger = KotlinLogging.logger {}
class NelsonNumber : INumberTypeOperation {
	// The number 111 is sometimes called a Nelson. Multiples of 111 are called a
	// double Nelson (222), triple Nelson (333)
	// 111,222,333,444,555,666,777
	override fun execute(input: Long, order: Long): Boolean {
		val status = input % 111 == 0L
		log(status, input)
		return status
	}

	private fun log(status: Boolean, input: Long) {
		if (status) {
			logger.debug("An Nelson Number {}", input)
		} else {
			logger.debug("Not an Nelson Number {}", input)
		}
	}
}